src/ipc/ipc.c

```diff
@@ -267,7 +267,7 @@ int ipc_buffer_new(struct ipc *ipc, struct sof_ipc_buffer *desc)
 	buffer = buffer_new(desc);
 	if (buffer == NULL) {
 		trace_ipc_error("ipc_buffer_new() error: buffer_new() failed");
-		rfree(ibd);
```

Comment:
At this point ibd is NULL, because of the if statement few lines above.

Reply:
@akloniex there's a similar one on line 276 as well?

Reply:
@ranj063 around line 276, we should call buffer_free(), instead rfree(buffer), that's a mem leak.
If allocation of ibd fails, then _properly_ free the buffer that was allocated with buffer_new at line 267, and abandon whole operation.
I suppose that I could add that as separate patch to this PR.